powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/configuration/NextStepServerConfiguration.java

```diff
@@ -252,4 +305,34 @@ public Audit audit() {
         return auditFactory.getAudit();
     }
 
+    /**
+     * Configure LDAP context source
+     * @return LDAP context source.
+     */
+    @ConditionalOnProperty(prefix = "powerauth.nextstep.ldap", name = "enabled", havingValue = "true")
+    @Bean
+    public LdapContextSource contextSource() {
+        LdapContextSource cs = new LdapContextSource();
+        cs.setUrl(ldapUrl);
+        cs.setBase(ldapBase);
+        if (!managerDn.isBlank()) cs.setUserDn(managerDn);
+        if (!managerPassword.isBlank()) cs.setPassword(managerPassword);
+        cs.setPooled(ldapPooled);
+        cs.setAnonymousReadOnly(ldapAnonymousReadOnly);
+
+        Hashtable<String, Object> env = new Hashtable<>();
```

Comment:
Uf, `Hastable` is synchronized. Moreover, the variable type should be rather an interface. Let's write something like
```suggestion
        final Map<String, Object> env = Map.of(...
```

Reply:
Yes, we can have the final Map....
But the map is actually copied to a Hashtable. So the adding to the map is more visually clear way than Map.of. 
If the LdapContextSource had a build this could be avoided.

Reply:
Nitpick: It is not a reason to reject this PR, but to clear my ideas. I think that you are writing about a method
```java
public void setBaseEnvironmentProperties(Map<String, Object> baseEnvironmentProperties) {
    this.baseEnv = new Hashtable(baseEnvironmentProperties);
}
```
It creates `Hashtable` internally, and we do not care; moreover, the constructor creates a copy by `putAll(t)`.
Hashtable is synchronized, considered legacy, and always a red flag for me. If synchronization is important in this case (I doubt it), prefer `ConcurrentHashMap`, otherwise use just `HashMap`.